probdists/Triangulardistribution.py

```diff
@@ -0,0 +1,199 @@
+                    (self.a * self.b) - (self.a * self.mode) - \
+                    (self.b * self.mode)
+        variance = summation / 18
+        self.stdev = math.sqrt(variance)
+
+        return round(self.stdev, round_to)
+
+    def replace_stats_with_data(self):
+        """ Method to calculate a, b, mode from the data set
+
+        Args:
+            None
+
+        Returns:
+            float: a, the minimum value
+            float: b, the maximum value
+            float: mode, the mode of the dataset
+        """
+        if not self.data:
+            # Use default values
+            self.a, self.b, self.mode = 0, 1, 0.5
+        else:
+            self.a = min(self.data)
+            self.b = max(self.data)
+            self.calculate_mode()
+
+        return self.a, self.b, self.mode
+
+    def calculate_mode(self, round_to=2):
+        """
+        Calculates the mode of a dataset
+        If no single mode, it will approximate the mode using the mean
+
+        Args:
+            round_to (int): Round the mode value. [Default value: 2]
+
+        Returns:
+            float: mode of data
+        """
+        data = Counter(self.data)
+        dict_data = dict(data)
+
+        # Create list of modes from data
+        mode = [k for k, v in dict_data.items() if v == max(list(data.values()))]
+
+        if len(mode) == 1:
+            self.mode = mode[0]
+        else:
+            # Multiple modes
+            # Approximates mode using mean
+            self.mode = sum(self.data) / len(self.data)
+        return round(self.mode, round_to)
+
+    def calculate_pdf(self, x, round_to=2):
+        """
+        Probability density function calculator for the Triangular distribution.
+
+        Args:
+            x (float): point for calculating the probability density function
+            round_to (int): Round the pdf value. [Default value: 2]
+
+        Returns:
+            float: probability density function
+        """
+        value = 0  # default value for when x < min or x > max
+        if self.a <= x <= self.mode:
+            value = (2 * (x - self.a)) / (
+                    (self.b - self.a) * (self.mode - self.a))
+        elif self.mode < x <= self.b:
+            value = (2 * (self.b - x)) / (
+                    (self.b - self.a) * (self.b - self.mode))
+
+        self.pdf = value
+        return round(self.pdf, round_to)
+
+    def calculate_cdf(self, x, round_to=2):
+        """
+        Cumulative density function calculator for the Triangular distribution.
+
+        Args:
+            x (float): point for calculating the cumulative density function
+            round_to (int): Round the value. [Default value: 2]
+
+        Returns:
+            float: cumulative density function output
+        """
+        if x < self.a:
+            value = 0
+        elif self.a <= x <= self.mode:
+            num = ((x - self.a) ** 2)
+            den = (self.b - self.a) * (self.mode - self.a)
```

Comment:
how do we plan to handle division by zero here?

Reply:
Calculating the CDF when min = mode becomes difficult and would cause zero division. Working on a workaround...any ideas?